src/Features/LanguageServer/Protocol/Handler/InlineCompletions/InlineCompletionsHandler.ParsedXmlSnippet.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.InlineCompletions;
+
+internal partial class InlineCompletionsHandler
```

Comment:
Do you typically utilize partial classes heavily in Roslyn? They typically fall over in some source linked debug scenarios + make understanding implementation details a bit more tedious. Would it be awful to pull all the XML snippet tech into a service that the primary InlineCompletionsHandler utilizes?

Reply:
partial classes are not uncommon in roslyn.  It sounds like source link debug scenarios need to be more resilient ;)

I don't mind either way really